### PR TITLE
Blood Stalker SPA

### PR DESCRIPTION
**Pull request by NickAragua**
Creates a new button in the firing phase that can be used to activate SPAs (this is mostly to avoid button clutter).

Implements the Blood Stalker SPA as per CamOps (+1 to weapons fire vs designated target, -2 to all others until target is destroyed or off the field permanently).

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/FiringDisplay.java [hunk 1]
```diff
@@ -1308,6 +1309,50 @@ private void doClearWeaponJam() {
...
+    
+    /**
+     * Worker function that determines if we can activate the "blood stalker" ability
+     */
+    private boolean canActivateBloodStalker() {
+        // can be activated if the entity can do it and haven't done it already 
+        // and the target is something that can be blood-stalked
+        return ce().canActivateBloodStalker() &&
+                (target != null) && (target.getTargetType() == Targetable.TYPE_ENTITY) &&
+                !attacks.stream().anyMatch(item -> item instanceof ActivateBloodStalkerAction);
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/FiringDisplay.java:1354]:
```suggestion
                attacks.stream().noneMatch(item -> item instanceof ActivateBloodStalkerAction);
```

File: megamek/src/megamek/server/Server.java [hunk 2]
```diff
@@ -30536,6 +30553,10 @@ public boolean accept(Entity entity) {
                     game.removeTurnFor(entity);
                     game.removeEntity(entityId, IEntityRemovalConditions.REMOVE_NEVER_JOINED);
                 }
+                
+                if (game.getPhase() != IGame.Phase.PHASE_LOUNGE) {
```
Comment (Windchild292) [inline megamek/src/megamek/server/Server.java:30557]:
```suggestion
                if (!getGame().getPhase().isLounge()) {
```

Comment (Windchild292) [reply megamek/src/megamek/server/Server.java:30557]: This is moved in #3220, which will merge shortly.

**Review by Windchild292 — COMMENTED** [review]
A few things first, plus the null alerts from LGTM.

**Comment by NickAragua** [comment]
The LGTM stuff is mostly red herrings, it thinks that just because I added a null check in one place it's needed in several places above which haven't experienced any problems to date.

**Review by Windchild292 — APPROVED** [review]
I'd prefer to preempt any potential detected issues, but no requirements to do so.
Approved for 0.49.6

**Pull request merged**